lib/krikri/enricher.rb

```diff
@@ -0,0 +1,132 @@
+module Krikri
+  ##
+  # A SoftwareAgent that runs enrichment processes.
+  #
+  # @example
+  #
+  #   To enrich records that were mapped by the mapping activity with ID 3:
+  #
+  #   # Define which enrichments are run, and thier parameters:
+  #   chain = {
+  #     'Krikri::Enrichments::StripHtml' => {
+  #       input_fields: [{sourceResource: :title}]
+  #     },
+  #     'Krikri::Enrichments::StripWhitespace' => {
+  #       input_fields: [{sourceResource: :title}]
+  #     }
+  #   }
+  #   Krikri::Enricher.enqueue({
+  #     generator_uri: 'http://ldp.local.dp.la/ldp/activity/3',
+  #     chain: chain
+  #   })
+  #
+  # @see Krikri::SoftwareAgent#enqueue
+  #
+  class Enricher
+    include SoftwareAgent
+
+    attr_reader :chain, :generator_uri
+
+    def self.queue_name
+      :enrichment
+    end
+
+    def initialize(opts = {})
+      @generator_uri = RDF::URI(opts.fetch(:generator_uri))
+      @chain = deep_sym(opts.fetch(:chain) { {} })
+    end
+
+    def run(activity_uri = nil)
+      log :info, 'enricher is running'
+      # see TODO below
+      target_aggregations.each do |agg|
+        chain_enrichments!(agg)
+        agg << RDF::Statement(agg, RDF::PROV.wasGeneratedBy, activity_uri) if
+          activity_uri
+        agg.save
+      end
+    rescue => e
+      Rails.logger.error("Enrichment error: #{e.message}\n#{e.backtrace}")
```

Comment:
Yes, it would be best to continue on, like `Harvester` does.  When I was looking at `Mapper::Agent#run`, I probably overlooked the fact that the rescue is inside of the loop in that method.  I'll change this to catch the exception inside of the `each` proc.